packages/api/src/controllers/stream.ts

```diff
@@ -229,6 +234,7 @@ function activeCleanupOne(
   }
 
   setImmediate(async () => {
+    await cleanupSemaphore.wait();
```

Comment:
The "right" solution would be to have a separate connection pool for the cleanup work. This is a workaround. I wonder how difficult would it be to implement a separate connection pool for the cleanup.

Reply:
I've implemented the separate connection pool here! https://github.com/livepeer/studio/pull/2013/commits/59a67fde46ca0a622d89dfb84dc2462547aa2768

I'd recommend reviewing that commit separately, since I touched several different files (also moved webhooks, tasks and usage logic to the jobs conn pool).